### PR TITLE
Rework of the install process

### LINKED ISSUE #151: fetch binaries instead of embedding them in installer
The `get.notionjs.com` installation flow already has a wrapper script that fetches the installer, so it's really pointless to embed the binaries in the shell script itself. And the template expander script is committing some _heinous_ crimes against software engineering. We should simplify the install script to just be a simpler script that curls all the binaries it needs, and eliminate all the templating entirely.

### PR DESCRIPTION
**Pull request by mikrostew**
Closes #151

## Summary

This reworks the install process to download the binary archive instead of embedding the binaries in the install script, basically combining things from `boot-install.sh` and `install.sh.in`, and adding some new functionality.

## What is Supported

Installing the latest release, the same way it works now:

```
$ ./dev/unix/volta-install.sh
  Installing latest version of Volta (0.5.0)
    Checking for existing Volta installation
    Fetching archive for macOS, version 0.5.0
######################################################################## 100.0%
    Creating directory layout
  Extracting Volta binaries and launchers
x volta
x shim
x load.fish
x load.sh
    Creating symlinks and shims
     Editing user profile (/Users/mistewar/.bash_profile)
Warning: Your profile (/Users/mistewar/.bash_profile) already mentions Volta and has not been changed.

    Finished installation. Open a new terminal to start using Volta!
```

Installing a specific release version. This is useful to reproduce an error, or to install and test a new version that has been released, before updating the `latest-version` on the website.

```
$ ./dev/unix/volta-install.sh --version 0.5.0
  Installing Volta version 0.5.0
    Checking for existing Volta installation
    Fetching archive for macOS, version 0.5.0
######################################################################## 100.0%
    Creating directory layout
  Extracting Volta binaries and launchers
x volta
x shim
x load.fish
x load.sh
    Creating symlinks and shims
     Editing user profile (/Users/mistewar/.bash_profile)
Warning: Your profile (/Users/mistewar/.bash_profile) already mentions Volta and has not been changed.

    Finished installation. Open a new terminal to start using Volta!
```

Compiling and installing locally. This supports compiling in `dev` or `release` mode, and is now just one command:

```
$ ./dev/unix/volta-install.sh --dev
  Installing Volta locally after compiling
    Checking for existing Volta installation
    Building Volta for volta-0.5.0-macos
   Compiling volta-core v0.1.0 (/Users/mistewar/Dropbox/src/notion/crates/volta-core)
   Compiling volta v0.5.0 (/Users/mistewar/Dropbox/src/notion)
    Finished dev [unoptimized + debuginfo] target(s) in 27.91s
   Packaging the compiled binaries and shell scripts
a volta
a shim
a load.fish
a load.sh
   Completed release in file target/debug/volta-0.5.0-macos.tar.gz
    Creating directory layout
  Extracting Volta binaries and launchers
x volta
x shim
x load.fish
x load.sh
    Creating symlinks and shims
     Editing user profile (/Users/mistewar/.bash_profile)
    Finished installation. Open a new terminal to start using Volta!
```

```
$ ./dev/unix/volta-install.sh --release
  Installing Volta locally after compiling with '--release'
    Checking for existing Volta installation
    Building Volta for volta-0.5.0-macos
   Compiling volta-core v0.1.0 (/Users/mistewar/Dropbox/src/notion/crates/volta-core)
   Compiling volta v0.5.0 (/Users/mistewar/Dropbox/src/notion)
    Finished release [optimized] target(s) in 46.68s
   Packaging the compiled binaries and shell scripts
a volta
a shim
a load.fish
a load.sh
   Completed release in file target/release/volta-0.5.0-macos.tar.gz
    Creating directory layout
  Extracting Volta binaries and launchers
x volta
x shim
x load.fish
x load.sh
    Creating symlinks and shims
     Editing user profile (/Users/mistewar/.bash_profile)
Warning: Your profile (/Users/mistewar/.bash_profile) already mentions Volta and has not been changed.

    Finished installation. Open a new terminal to start using Volta!
```

## What is Not Supported

This installer does not support installing versions prior to 0.5.0, when the project was called Notion.

### CODE REVIEW

File: dev/unix/notion-install.sh [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+while [ $# -gt 0 ]
+do
+  arg="$1"
+
+  case "$arg" in
+    -h|--help)
+      usage
+      exit 0
+      ;;
+    --dev)
```
Comment (benblank) [inline dev/unix/notion-install.sh]: In `usage()` above, this is named `--debug`.

Comment (mikrostew) [reply dev/unix/notion-install.sh]: ah, good catch! I will change that

File: dev/unix/volta-install.sh [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+parse_openssl_version() {
+  local version_str="$1"
+
+  # array containing the SSL libraries that are supported
+  # would be nice to use a bash 4.x associative array, but bash 3.x is the default on OSX
+  SUPPORTED_SSL_LIBS=( 'OpenSSL' )
+
+  # use regex to get the library name and version
+  # typical version string looks like 'OpenSSL 1.0.1e-fips 11 Feb 2013'
+  if [[ "$version_str" =~ ^([^\ ]*)\ ([0-9]+\.[0-9]+\.[0-9]+) ]]
```
Comment (charlespierce) [inline dev/unix/volta-install.sh]: In a complete reversal of my previous comment, it turns out we actually don't care about the patch version, and only care about the major and minor, so we should just match to those (sorry).

Comment (mikrostew) [reply dev/unix/volta-install.sh]: Ah, right! I saw that change in #409 and forgot to change it in this PR - will update that to match the current behavior of `boot-install.sh`

**Pull request merged**